### PR TITLE
81 supplier plugin health and store usage after revert

### PR DESCRIPTION
**Pull request by JohannesDaniel**
### Description
This PR includes some refactorings for comments in https://github.com/opensearch-project/opensearch-learning-to-rank-base/pull/84

### CODE REVIEW

**Comment by JohannesDaniel** [comment]
@ylwu-amzn new PR after revert.

Fixed 
* code in comment in StoreUtils
* var naming in PluginHealthStatusSupplier
* license in PluginHealthStatusSupplier

**Comment by ylwu-amzn** [comment]
I added some comments to the old PR, can you help address those comments first ? https://github.com/opensearch-project/opensearch-learning-to-rank-base/pull/84
I can take another look once that comments addressed

File: src/main/java/org/opensearch/ltr/stats/suppliers/utils/StoreUtils.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+            if (featureSet != null && featureSet.containsKey(FEATURES_KEY) && featureSet.containsKey(FEATURE_SET_NAME_KEY)) {
+                return Optional.of(featureSet);
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    public long getModelCount(String storeName) {
```
Comment (JohannesDaniel) [inline src/main/java/org/opensearch/ltr/stats/suppliers/utils/StoreUtils.java:100]: Removed comment as suggested in
https://github.com/opensearch-project/opensearch-learning-to-rank-base/pull/84#discussion_r1880999552

File: src/main/java/org/opensearch/ltr/stats/suppliers/PluginHealthStatusSupplier.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
```
Comment (JohannesDaniel) [inline src/main/java/org/opensearch/ltr/stats/suppliers/PluginHealthStatusSupplier.java:15]: Added license header as suggested in https://github.com/opensearch-project/opensearch-learning-to-rank-base/pull/84#discussion_r1881000597

File: src/main/java/org/opensearch/ltr/stats/suppliers/PluginHealthStatusSupplier.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    private String getAggregateStoresStatus() {
+        List<String> storeNames = storeUtils.getAllLtrStoreNames();
+        return storeNames.stream()
+                .map(storeUtils::getLtrStoreHealthStatus)
+                .reduce(STATUS_GREEN, this::combineStatuses);
+    }
+
+    private String combineStatuses(String status1, String status2) {
+        if (status2 == null || STATUS_RED.equals(status1) || STATUS_RED.equals(status2)) {
```
Comment (JohannesDaniel) [inline src/main/java/org/opensearch/ltr/stats/suppliers/PluginHealthStatusSupplier.java:60]: Made method params more meaningful as suggested in https://github.com/opensearch-project/opensearch-learning-to-rank-base/pull/84#discussion_r1881003199

**Review by rithin-pullela-aws — APPROVED** [review]

File: src/main/java/org/opensearch/ltr/stats/suppliers/StoreStatsSupplier.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
+package org.opensearch.ltr.stats.suppliers;
```
Comment (ylwu-amzn) [inline src/main/java/org/opensearch/ltr/stats/suppliers/StoreStatsSupplier.java:1]: Please add license header for all files

File: src/test/java/org/opensearch/ltr/stats/suppliers/PluginHealthStatusSupplierTests.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
+package org.opensearch.ltr.stats.suppliers;
```
Comment (ylwu-amzn) [inline src/test/java/org/opensearch/ltr/stats/suppliers/PluginHealthStatusSupplierTests.java:1]: Add license header

File: src/main/java/org/opensearch/ltr/stats/suppliers/PluginHealthStatusSupplier.java (hunk 3 above)
Comment (ylwu-amzn) [reply src/main/java/org/opensearch/ltr/stats/suppliers/PluginHealthStatusSupplier.java:60]: Thanks, `status2` will be from `getLtrStoreHealthStatus` method, which seems never returns null value, right ? Is it necessary to check `status2 == null`?

**Pull request merged**